tests/integration/targets/aci_ip_sla_monitoring_policy/tasks/main.yml

```diff
@@ -0,0 +1,256 @@
+    tenant: ansible_tenant
+    sla_policy: ansible_http_sla
+    sla_type: icmp
+    sla_port: 8080
+    state: present
+  ignore_errors: true
+  register: icmp_with_sla_port
+
+- name: Create ICMP Monitoring Policy with request_data_size
+  cisco.aci.aci_ip_sla_monitoring_policy:
+    <<: *aci_info
+    tenant: ansible_tenant
+    sla_policy: ansible_http_sla
+    sla_type: icmp
+    request_data_size: 2000
+    state: present
+  ignore_errors: true
+  register: icmp_with_request_data_size
+
+- name: Verify Errors
+  ansible.builtin.assert:
+    that:
+    - icmp_with_sla_port is failed
+    - icmp_with_request_data_size is failed
+    - icmp_with_sla_port.msg == "Setting 'sla_port' is not allowed when 'sla_type' is not set to 'tcp'."
+    - icmp_with_request_data_size.msg == "Setting 'request_data_size' is not allowed when 'sla_type' is not set to 'http."
+
+# MODIFY SLA POLICY
+- name: Convert ICMP SLA policy to TCP
+  cisco.aci.aci_ip_sla_monitoring_policy:
+    <<: *aci_info
+    tenant: ansible_tenant
+    sla_policy: ansible_sla
+    sla_type: tcp
+    sla_port: 8080
+    frequency: 20
+    multiplier: 5
+    state: present
+  register: update_tcp_sla
+
+- name: Verify SLA update
+  ansible.builtin.assert:
+    that:
+    - update_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.dn == "uni/tn-ansible_tenant/ipslaMonitoringPol-ansible_sla"
+    - update_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.name == "ansible_sla"
+    - update_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaDetectMultiplier == "5"
+    - update_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaFrequency == "20"
+    - update_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaPort == "8080"
+    - update_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaType == "tcp"
+
+# QUERY IP SLA POLICY
+- name: Query IP SLA monitor
+  cisco.aci.aci_ip_sla_monitoring_policy:
+    <<: *aci_info
+    tenant: ansible_tenant
+    sla_policy: ansible_sla
+    state: query
+  register: query_tcp_sla
+
+- name: Verify SLA query
+  ansible.builtin.assert:
+    that:
+    - query_tcp_sla is not changed
+    - query_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.dn == "uni/tn-ansible_tenant/ipslaMonitoringPol-ansible_sla"
+    - query_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.name == "ansible_sla"
+    - query_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaDetectMultiplier == "5"
+    - query_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaFrequency == "20"
+    - query_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaPort == "8080"
+    - query_tcp_sla.current.0.fvIPSLAMonitoringPol.attributes.slaType == "tcp"
+
+# QUERY ALL SLA POLICIES
+- name: Query all IP SLA monitor policies
+  cisco.aci.aci_ip_sla_monitoring_policy:
+    <<: *aci_info
+    state: query
+  register: query_sla_all
+
+- name: Verify all SLA query
+  ansible.builtin.assert:
+    that:
+    - query_sla_all is not changed
+    - query_sla_all.current | length > 1
```

Comment:
You can add the query_sla_all verification with the exact value check something like this: https://github.com/CiscoDevNet/ansible-aci/pull/723/files#diff-9d4dbc463b458c73dace4b59bcbd2707668ced9c97684aa88d0b7840b4f3329cR284-R287